compiler/ast/src/types/type_.rs

```diff
@@ -68,6 +80,76 @@ pub enum Type {
 }
 
 impl Type {
+    /// Are the types considered equal as far as the Leo user is concerned?
+    ///
+    /// In particular, any comparison involving an `Err` is `true`,
+    /// composite types are resolved to the current program if not specified,
```

Comment:
"composite types are resolved to the current program" seems inconsistent with "Composite types are considered equal..." and with the implementation.